From dde5205d60d6783ebc4ce7df76318130461ec753 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fr=C3=A9d=C3=A9ric=20Pierret=20=28fepitre=29?=
 <frederic.pierret@qubes-os.org>
Date: Mon, 1 Nov 2021 18:35:52 +0100
Subject: [PATCH] pc_piix: suppress vmdesc

Skip waiting for VMDESC state section, it isn't sent when resuming Xen domain.

Adapted from https://github.com/QubesOS/qubes-vmm-xen-stubdom-linux/blob/2a814bd1edaf549ef9252eb6747aa6137abf9831/qemu/patches/suppress_vmdesc.patch

Author: marmarek

Contributors: fepitre
---
 hw/i386/pc_piix.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index 20962c34e..2cc13e654 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -402,6 +402,7 @@ static void pc_xen_hvm_init(MachineState *machine)
     }
 
     pc_xen_hvm_init_pci(machine);
+    machine->suppress_vmdesc = true;
     pci_create_simple(pcms->bus, -1, "xen-platform");
 }
 #endif
-- 
2.31.1

